src/test/java/org/generationcp/middleware/service/impl/inventory/PlantingServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,328 @@
+		withdrawalInstruction.setGroupTransactions(true);
+
+		final Map<String, PlantingRequestDto.WithdrawalInstruction> withdrawalInstructionMap = new HashMap<>();
+		withdrawalInstructionMap.put(unitName, withdrawalInstruction);
+		plantingRequestDto.setWithdrawalsPerUnit(withdrawalInstructionMap);
+
+		final PlantingRequestDto.LotEntryNumber lotEntryNumber = new PlantingRequestDto.LotEntryNumber();
+		lotEntryNumber.setEntryNo(entryNo);
+		lotEntryNumber.setLotId(lot.getId());
+		plantingRequestDto.setLotPerEntryNo(Collections.singletonList(lotEntryNumber));
+
+		try {
+			this.plantingService.savePlanting(userId, studyId, observationDatasetId, plantingRequestDto, TransactionStatus.CONFIRMED);
+		} catch (final MiddlewareRequestException e) {
+			assertThat(e.getErrorCode(), equalTo("planting.confirmed.transactions.found"));
+		}
+	}
+
+	@Test
+	public void test_SavePlanting_NotEnoughInventory_ThrowsException() {
+		final Integer gid = Integer.valueOf(
+			experiments.get(0).getFactors().getVariables().stream().filter(v -> v.getVariableType().getLocalName().equals("GID"))
+				.findFirst().get().getValue());
+		final Integer entryNo = Integer.valueOf(
+			experiments.get(0).getFactors().getVariables().stream().filter(v -> v.getVariableType().getLocalName().equals("ENTRY_NO"))
+				.findFirst().get().getValue());
+		final Lot lot = createLot(gid);
+		this.createTransaction(lot, TransactionType.DEPOSIT, TransactionStatus.CONFIRMED, 100D);
+
+		final PlantingRequestDto plantingRequestDto = new PlantingRequestDto();
+		final SearchCompositeDto<ObservationUnitsSearchDTO, Integer> searchCompositeDto = new SearchCompositeDto<>();
+		searchCompositeDto.setItemIds(Collections.singleton(experiments.get(0).getId()));
+		plantingRequestDto.setSelectedObservationUnits(searchCompositeDto);
+
+		final PlantingRequestDto.WithdrawalInstruction withdrawalInstruction = new PlantingRequestDto.WithdrawalInstruction();
+		withdrawalInstruction.setWithdrawAllAvailableBalance(false);
+		withdrawalInstruction.setGroupTransactions(true);
+		withdrawalInstruction.setWithdrawalAmount(200D);
+
+		final Map<String, PlantingRequestDto.WithdrawalInstruction> withdrawalInstructionMap = new HashMap<>();
+		withdrawalInstructionMap.put(unitName, withdrawalInstruction);
+		plantingRequestDto.setWithdrawalsPerUnit(withdrawalInstructionMap);
+
+		final PlantingRequestDto.LotEntryNumber lotEntryNumber = new PlantingRequestDto.LotEntryNumber();
+		lotEntryNumber.setEntryNo(entryNo);
+		lotEntryNumber.setLotId(lot.getId());
+		plantingRequestDto.setLotPerEntryNo(Collections.singletonList(lotEntryNumber));
+
+		try {
+			this.plantingService.savePlanting(userId, studyId, observationDatasetId, plantingRequestDto, TransactionStatus.CONFIRMED);
+		} catch (final MiddlewareRequestException e) {
+			assertThat(e.getErrorCode(), equalTo("planting.not.enough.inventory"));
+		}
+	}
+
+	@Test
+	public void test_SavePlanting_GroupTransactions_ThrowsException() {
```

Comment:
the last two don't throw an exception right?

Reply:
Fixed names, also added assert by quantity